src/Shared.CLI/Helpers/OutputBuilder/SarifOutputBuilder.cs

```diff
@@ -27,7 +27,7 @@ public SarifOutputBuilder(SarifVersion version)
         /// <returns>Location list with single location object</returns>
         public static List<Location> BuildPurlLocation(PackageURL purl)
         {
-            BaseProjectManager? projectManager = ProjectManagerFactory.CreateProjectManager(purl, null);
+            BaseProjectManager? projectManager = ProjectManagerFactory.GetProjectManager(purl, null);
```

Comment:
Why rename Get instead of Create? Get implies to me that the object is somehow a singleton/cached. But this method always eventually ctors (at least for the default managers, your `CreatePackageManager` method could return a singleton if you desired).

Reply:
I switched it to `ConstructPackageManager` and added the `overrideManagers` dict and the `destinationDirectory` as optional parameters. That make sense?